app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageStocksSearchByKeywordTest.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../../../../vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Test/etc/testSchema.xsd">
+    <test name="AdminManageStocksSearchByKeywordTest">
+        <annotations>
+            <features value="Multi-Source Inventory"/>
+            <stories value="Search By Keyword in Grid"/>
+            <title value="Search by Keyword on the Manage Stock grid"/>
+            <description value="You should be able to Search by Keyword from the Manage Stock grid."/>
+            <testCaseId value="515"/>
+            <group value="msi"/>
+        </annotations>
+        <before>
+            <createData entity="_defaultStock" stepKey="createStock1"/>
+            <createData entity="_defaultStock" stepKey="createStock2"/>
+            <createData entity="_defaultStock" stepKey="createStock3"/>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+        </before>
+        <after>
+            <actionGroup ref="logout" stepKey="logoutOfAdmin"/>
+        </after>
+
+        <amOnPage url="{{ManageStock.url}}" stepKey="amOnTheStockListPage"/>
+        <waitForPageLoad stepKey="waitForPageLoad1"/>
+
+        <actionGroup ref="AdminGridSearchByKeyword" stepKey="searchByKeyword">
+            <argument name="value" value="$$createStock3.general[name]$$"/>
```

Comment:
In scenario described put only piece of Stock name and exactly number "4" to search by keyword field and verify that "Test Stock 4" appears.
```
9. Put in "Search by keyword" field value "4"
10. verify that search returned "Test Stock 4" in grid
11. Then Search by keyword worked properly in Manage Stocks grid
```